nodebuilder/share/module.go

```diff
@@ -52,13 +52,21 @@ func ConstructModule(tp node.Type, cfg *Config, options ...fx.Option) fx.Option
 				func(path node.StorePath, ds datastore.Batching) (*eds.Store, error) {
 					return eds.NewStore(string(path), ds)
 				},
-				fx.OnStart(func(ctx context.Context, eds *eds.Store) error {
-					return eds.Start(ctx)
+				fx.OnStart(func(ctx context.Context, store *eds.Store) error {
+					err := store.Start(ctx)
+					if err != nil {
+						return err
+					}
+
+					return ensureEmptyCARExists(ctx, store)
```

Comment:
Let's do it in Invoke as we do for blockservice? We won't need a forceful Invoke this way, either.

Reply:
Can't, because invokes happen before start is successful. If we invoke this, it will hang forever because the EDS store is not started